library/src/main/java/com/esafirm/rxdownloader/RxDownloader.java

```diff
@@ -47,44 +48,108 @@ private DownloadManager getDownloadManager() {
+    public Single<String> download(@NonNull String url,
+                                             @NonNull String filename,
+                                             boolean showCompletedNotification) {
         return download(url, filename, DEFAULT_MIME_TYPE, showCompletedNotification);
     }
 
-    public Observable<String> download(@NonNull String url,
-                                       @NonNull String filename,
-                                       @NonNull String mimeType,
-                                       boolean showCompletedNotification) {
+    public Single<String> download(@NonNull String url,
+                                             @NonNull String filename,
+                                             @NonNull String mimeType,
+                                             boolean showCompletedNotification) {
         return download(createRequest(url, filename, null,
                 mimeType, true, showCompletedNotification));
     }
 
-    public Observable<String> download(@NonNull String url,
-                                       @NonNull String filename,
-                                       @NonNull String destinationPath,
-                                       @NonNull String mimeType,
-                                       boolean showCompletedNotification) {
+    public Single<String> download(@NonNull String url,
+                                             @NonNull String filename,
+                                             @NonNull String destinationPath,
+                                             @NonNull String mimeType,
+                                             boolean showCompletedNotification) {
         return download(createRequest(url, filename, destinationPath,
                 mimeType, true, showCompletedNotification));
     }
 
-    public Observable<String> downloadInFilesDir(@NonNull String url,
-                                                 @NonNull String filename,
-                                                 @NonNull String destinationPath,
-                                                 @NonNull String mimeType,
-                                                 boolean showCompletedNotification) {
+    public Single<String> downloadInFilesDir(@NonNull String url,
+                                                       @NonNull String filename,
+                                                       @NonNull String destinationPath,
+                                                       @NonNull String mimeType,
+                                                       boolean showCompletedNotification) {
         return download(createRequest(url, filename, destinationPath,
                 mimeType, false, showCompletedNotification));
     }
 
-    public Observable<String> download(DownloadManager.Request request) {
+    public Single<String> download(DownloadManager.Request request) {
```

Comment:
For simple download i think we should retain the current approach because it's faster (don't have to wait for the tick)

Reply:
Makes sense. Just wonder if race conditions might occur because one might finish before the other. Perhaps, not. I'll re-add the `BroadcastReceiver` and listen to finish events in both the ticking thread and the receiver.

Reply:
Are you happy with changing the single download calls to a `Single` or shall I keep it as an `Observable`?

Reply:
I think `Single` is the right type for this 👍

Reply:
> Just wonder if race conditions might occur because one might finish before the other.

What do you mean by this?

Reply:
What I mean, is if the ticking thread gets the progress completed state first, then would need to stop the broadcast receiver at that point and vice versa else we may get two completed events. Ooor, isn't there an Rx operator that would do it, like `combine(Completable a,Completable b)`?